torch/autograd/gradcheck.py

```diff
@@ -806,41 +886,88 @@ def wrapped_fn(*inputs):
         return wrapped_fn
     return apply_to_c_outs(fn, torch.real), apply_to_c_outs(fn, torch.imag)
 
+def _real_and_imag_input(fn):
+    # returns new functions that take real inputs instead of complex inputs and compute fn(x + 0 * 1j)
+    # and f(x * 1j).
+    def apply_to_c_inps(fn, fn_to_apply):
+        def wrapped_fn(*inputs):
+            new_inputs = (fn_to_apply(inp) if inp.is_complex() else inp for inp in inputs)
+            return _as_tuple(fn(*new_inputs))
+        return wrapped_fn
+    return apply_to_c_inps(fn, lambda x: x + 0 * 1j), apply_to_c_inps(fn, lambda x: x * 1j)
```

Comment:
This is an interesting way of converting number to complex `x + 0 * 1j`, did you just do this for symmetry (because the other lambda had `x * 1j`)?

Reply:
Not, that's because that way, I don't need to worry about real -> float type promotion for float32 vs float64 :D And so I don't need a manual mapping of type promotions :D